Add logic to validate project and robot information in ns controller

hanlins opened the pull request:
As the first step moving project & robot validation logic from PSB ctrl to ns controller, duplicating the logic into ns controller.

hanlins committed:
Add logic to validate project and robot information in ns controller

As the first step moving proj/robot validation logic from PSB ctrl to ns
ctrl, duplicating the logic to ns controller.

hanlins committed:
Update PSB object schema

Signed-off-by: Hanlin Shi <shihanlin9@gmail.com>

hanlins committed:
Remove logics for robot account creation in PSB controller

Signed-off-by: Hanlin Shi <shihanlin9@gmail.com>

lubronzhan commented on controllers/namespace_controller.go:
```diff
@@ -158,3 +198,28 @@ func (r *NamespaceReconciler) getNewBindingCR(ns string, harborCfg string, sa st
+
+func (r *NamespaceReconciler) validateProjectRobot(proj, robot string) error {
+	robotID, err := strconv.ParseInt(robot, 10, 64)
+	if err == nil {
```
It should be error != nil

lubronzhan committed:
Fix bugs
Create missing harbor client
Store projectid and robotid in psb
If specified project doesn't exist, should error out

hanlins commented:
@lubronzhan thanks! LGTM.
